android-kotlin/app/src/main/java/io/metropolislab/samplearchitecture/presentation/green/GreenViewModel.kt

```diff
@@ -3,10 +3,9 @@ package io.metropolislab.samplearchitecture.presentation.green
 import io.metropolislab.samplearchitecture.architecture.BaseViewModel
 import java.lang.ref.WeakReference
 
-class GreenViewModel : BaseViewModel() {
-    var navigator: WeakReference<GreenNavigator>? = null
+class GreenViewModel(val navigator: WeakReference<GreenNavigator>) : BaseViewModel() {
 
     fun onBackPressed() {
-        navigator?.get()?.greenDone()
+        navigator.get()?.greenDone()
```

Comment:
👏  much better

Reply:
By the way, after giving a thought, I'm not sure we need to pass a weak reference to the navigator. Afterall  this will prevent the Corrdinator from not being garbage collected, but, I guest that if the Coordinator is still alive and can't be marked as "eligible" for collection, what it's wrong is that there's a VM still alive. Well, I think my explanation is not very clever . Better we talk about this on Monday.